pkg/client/selfhosted/path_test.go

```diff
@@ -0,0 +1,113 @@
+package selfhosted
+
+import "testing"
+
+func TestIsHost(t *testing.T) {
+	tests := map[string]struct {
+		host  string
+		expIs bool
+	}{
+		"an empty host should be false": {
+			host:  "",
+			expIs: false,
+		},
+		"random string should be false": {
+			host:  "foobar",
+			expIs: false,
+		},
+		"random string with dots should be false": {
+			host:  "foobar.foo",
+			expIs: false,
+		},
+		"just docker.io should be false": {
+			host:  "docker.io",
+			expIs: false,
+		},
+		"just docker.com should be false": {
+			host:  "docker.com",
+			expIs: false,
+		},
+		"docker.com with random sub domains should be false": {
+			host:  "foo.bar.docker.com",
+			expIs: false,
+		},
+		"docker.io with random sub domains should be false": {
+			host:  "foo.bar.docker.io",
+			expIs: false,
+		},
+		"docker.repositories.yourdomain.ext should be true": {
+			host:  "docker.repositories.yourdomain.ext",
+			expIs: true,
+		},
+		"docker.repositories.yourdomain.ext/testing should be false": {
+			host:  "docker.repositories.yourdomain.ext/testing",
+			expIs: false,
+		},
+		"docker.repositories.yourdomain.ext/artifactory/v2 should be false": {
+			host:  "docker.repositories.yourdomain.ext/artifactory/v2",
+			expIs: false,
+		},
+		"docker.comfoo should be false": {
+			host:  "docker.iofoo",
+			expIs: false,
+		},
+		"docker.iofoo should be false": {
+			host:  "ocker.iofoo",
+			expIs: false,
+		},
+	}
```

Comment:
Some tests with `expIs == true`

Reply:
Any thoughts on how, because there could/should be only one true.

Reply:
This test is only for the host. You are referring to the TestRepoImage part. These are split.